src/did-change-watched-files.jl

```diff
@@ -0,0 +1,93 @@
+using TOML
+
+const DID_CHANGE_WATCHED_FILES_REGISTRATION_ID = "jetls-did-change-watched-files"
+const DID_CHANGE_WATCHED_FILES_REGISTRATION_METHOD = "workspace/didChangeWatchedFiles"
+
+function did_change_watched_files_registration()
+    Registration(;
+        id = DID_CHANGE_WATCHED_FILES_REGISTRATION_ID,
+        method = DID_CHANGE_WATCHED_FILES_REGISTRATION_METHOD,
+        registerOptions = DidChangeWatchedFilesRegistrationOptions(;
+            watchers = [
+                FileSystemWatcher(;
+                    globPattern = "**/JETLSConfig.toml",
+                    kind = WatchKind.Create | WatchKind.Change | WatchKind.Delete),
+            ]))
+end
+
+# For dynamic registrations during development
+# unregister(currently_running, Unregistration(;
+#     id=DID_CHANGE_WATCHED_FILES_REGISTRATION_ID,
+#     method=DID_CHANGE_WATCHED_FILES_REGISTRATION_METHOD))
+# register(currently_running, did_change_watched_files_registration())
+
+function initialize_config!(server::Server)
+    if !isdefined(server.state, :root_path)
+        if JETLS_DEV_MODE
+            @info "server.state.root_path is not defined, skipping add JETLSConfig.toml watcher"
+        end
+        return
+    end
+    config_path = joinpath(server.state.root_path, "JETLSConfig.toml")
+    push!(server.state.config_manager.watching_files, config_path)
+    load_config!(server, config_path) do actual_config, latest_config, key_path, v
```

Comment:
```suggestion
    load_config!(server, config_path) do actual_config, latest_config, key, v
```
Maybe we can make `merge_config!` pass `key` only for the callback.

Reply:
Actually, We need full `key_path` here.

https://github.com/aviatesk/JETLS.jl/blob/34df7c63498da803d94b2b586a29c444ad133ce6/src/did-change-watched-files.jl#L71-L80